workbench-core/datasets/src/dataSetService.ts

```diff
@@ -187,23 +191,47 @@ export class DataSetService {
       endPointid: endPointId
     };
     try {
-      const targetDS = await this.getDataSet(dataSetId, authenticatedUser);
+      await this.getDataSet(dataSetId, authenticatedUser);
 
-      if (!_.find(targetDS.externalEndpoints, (ep) => ep === endPointId)) {
-        throw Boom.notFound(`'${endPointId}' not found on DataSet '${dataSetId}'.`);
+      const { data: permissionsData } = await this._authzPlugin.getAccessPermissions({
+        dataSetId,
+        identity,
+        identityType
+      });
+      if (!permissionsData.permissions.length) {
+        throw new NotAuthorizedError(
+          `${identityType} "${identity}" does not have permission to access dataSet "${dataSetId}.`
+        );
       }
 
       const endPoint = await this.getExternalEndPoint(dataSetId, endPointId, authenticatedUser);
-      if (!endPoint.endPointAlias || !endPoint.id) {
-        throw Boom.notFound('Endpoint has missing information');
+      if (!endPoint.endPointAlias) {
+        throw new InvalidEndpointError(
+          `Endpoint "${endPointId}" does not have an "endPointAlias" associated with it.`
+        );
       }
 
-      const response = this._generateMountObject(
-        endPoint.dataSetName,
-        endPoint.endPointAlias!,
-        endPoint.path,
-        endPoint.id!
+      const hasEndPointPermission = permissionsData.permissions.some(
+        ({ accessLevel }) => accessLevel === endPoint.accessLevel
       );
+
+      if (!hasEndPointPermission) {
+        throw new NotAuthorizedError(
+          `${identityType} "${identity}" does not have permission to access endpoint "${endPointId}.`
+        );
+      }
+
+      const response = {
+        data: {
+          mountObject: this._generateMountObject(
+            endPoint.dataSetName,
+            endPoint.endPointAlias!,
```

Comment:
Does anyone remember why we made `endpointAlias` optional? It seems like we're guaranteeing the compiler that it will be defined here, but that isnt the case right?

Reply:
Looks like that piece of history is no more, but seeing it at the end of the object, I assume it was added later in the public preview development cycle as a result of insufficient planning around access points and was added as optional for back compat.

Reply:
Would you be opposed to requiring it considering we are already making that assumption? Back compat shouldnt be a concern since we're still in beta